### PR TITLE
feat(expo)!: use Expo CLI when building for development in Xcode

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Related #21396 
- Use `npx expo start --dev-client` instead of `npx react-native start` when building a project from Xcode. This ensures we use the correct version of Metro regardless of where the project is started.
- It's unclear if building from Android Studio is supposed to trigger the bundler in development but I couldn't get it to do so, so this PR is iOS-only.

# How

- Copy over a bunch of the default scripts from `react-native/scripts` into the same locations in `expo/scripts` to make it easy to maintain this change and simple to migrate.

I decided to omit the following as it appeared to be unused (sets RCT_METRO_PORT):

```
source "$THIS_DIR/.packager.env"
```

I also dropped:
- Support for `$RCT_PACKAGER_LOGS_DIR`.
- The `Process terminated. Press <enter> to close the window` log that is presented after closing the dev server.
- Support for a custom React Native CLI config (this is not supported in Expo CLI).

# Test Plan

- TBD
- Building for development from Xcode causes a terminal window running `npx expo start` will open instead of the community version.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/@expo/config-plugins/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 ### 💡 Others
 
+- Update fixtures. ([#21397](https://github.com/expo/expo/pull/21397) by [@EvanBacon](https://github.com/EvanBacon))
```
Comment (EvanBacon) [inline packages/@expo/config-plugins/CHANGELOG.md:13]:
```suggestion
```

File: apps/bare-expo/ios/BareExpo.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -777,7 +777,7 @@
...
+			shellScript = "export RCT_METRO_PORT=\"${RCT_METRO_PORT:=8081}\"\necho \"export RCT_METRO_PORT=${RCT_METRO_PORT}\" > \"${SRCROOT}/../node_modules/react-native/scripts/.packager.env\"\nif [ -z \"${RCT_NO_LAUNCH_PACKAGER+xxx}\" ] ; then\n  if nc -w 5 -z localhost ${RCT_METRO_PORT} ; then\n    if ! curl -s \"http://localhost:${RCT_METRO_PORT}/status\" | grep -q \"packager-status:running\" ; then\n      echo \"Port ${RCT_METRO_PORT} already in use, packager is either not running or not running correctly\"\n      exit 2\n    fi\n  else\n    open \"$SRCROOT/../node_modules/expo/scripts/launchPackager.command\" || echo \"Can't start packager automatically\"\n  fi\nfi\n";
```
Comment (Kudo) [inline apps/bare-expo/ios/BareExpo.xcodeproj/project.pbxproj:780]: do you want to update this as the change with BareExpoDetox target?
e.g. the 
```
if [[ -f "$PODS_ROOT/../.xcode.env" ]]; then
  source "$PODS_ROOT/../.xcode.env"
fi
if [[ -f "$PODS_ROOT/../.xcode.env.local" ]]; then
  source "$PODS_ROOT/../.xcode.env.local"
fi

```

File: packages/@expo/cli/CHANGELOG.md [hunk 3]
```diff
@@ -20,6 +20,8 @@
 
 ### 💡 Others
 
+- Update fixtures. ([#21397](https://github.com/expo/expo/pull/21397) by [@EvanBacon](https://github.com/EvanBacon))
+- Update fixtures. ([#21397](https://github.com/expo/expo/pull/21397) by [@EvanBacon](https://github.com/EvanBacon))
```
Comment (Kudo) [inline packages/@expo/cli/CHANGELOG.md:24]:
```suggestion
- Update fixtures. ([#21397](https://github.com/expo/expo/pull/21397) by [@EvanBacon](https://github.com/EvanBacon))
```

Comment (Kudo) [reply packages/@expo/cli/CHANGELOG.md:24]: duplicated changelog

**Pull request merged**